pkg/util/pod/pod.go

```diff
@@ -56,3 +63,38 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+var (
+	errLabelNotFound = errors.New("label not found")
+	errInvalidUInt   = errors.New("invalid unsigned integer")
+	errValidation    = errors.New("validation error")
+)
+
+func ReadUIntFromLabel(obj client.Object, labelKey string) (*int, error) {
+	return ReadUIntFromLabelBelowBound(obj, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelBelowBound(obj client.Object, labelKey string, bound int) (*int, error) {
+	value, found := obj.GetLabels()[labelKey]
+	kind := obj.GetObjectKind().GroupVersionKind().Kind
+	if !found {
+		return nil, fmt.Errorf("%w: no label %q for %s %q", errLabelNotFound, labelKey, kind, klog.KObj(obj))
+	}
+	intValue, err := readUIntFromStringBelowBound(value, bound)
+	if err != nil {
+		return nil, fmt.Errorf("incorrect label value %q for %s %q: %w", value, kind, klog.KObj(obj), err)
+	}
+	return intValue, nil
+}
+
+func readUIntFromStringBelowBound(value string, bound int) (*int, error) {
+	uintValue, err := strconv.ParseUint(value, 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errInvalidUInt, err.Error())
+	}
+	intValue := int(uintValue)
+	if intValue > bound {
```

Comment:
```suggestion
	if uintValue > uint64(bound) {
```
This is to prevent overflow if uintValue was bigger than maxInt